packages/debug/src/browser/debug-session.tsx

```diff
@@ -302,22 +311,27 @@ export class DebugSession implements CompositeTreeElement {
             await this.disconnect(restart);
         }
     }
+
     protected async disconnect(restart?: boolean): Promise<void> {
-        try {
-            await this.sendRequest('disconnect', { restart });
-        } catch (reason) {
-            this.fireExited(reason);
-            return;
-        }
-        const timeout = 500;
-        if (!await this.exited(timeout)) {
-            this.fireExited(new Error(`timeout after ${timeout} ms`));
-        }
+        const TIMEOUT_MS = 1000;
```

Comment:
why was it increased from 500?

Reply:
Previously we ran await this.sendRequest('disconnect', { restart }); and after it completed we ran if (!await this.exited(timeout)) { - so in reality we were waiting longer then 500ms to check if exited was received.

Now we are invoking this.sendRequest and setTimeout together - therefore I thought it would be more prudent to add a little more to timeout. I don't want that sendRequest will succeed but timeout will be thrown before it - better in my mind to let it some more rope. Also, I tested it with 500/1000ms and I think it worked better with 1000.